modules/features2d/src/annoy/kissrandom.h

```diff
@@ -0,0 +1,120 @@
+#ifndef ANNOY_KISSRANDOM_H
+#define ANNOY_KISSRANDOM_H
+
+#if defined(_MSC_VER) && _MSC_VER == 1500
+typedef unsigned __int32    uint32_t;
+typedef unsigned __int64    uint64_t;
+#else
+#include <stdint.h>
+#endif
+
+namespace Annoy {
```

Comment:
use namespace 'cv::Annoy' instead of 'Annoy' to reduce possibility of name conflicts when OpenCV is used together with Annoy in some application